docs/developer/decision-records/2022-03-11-umbrella-issues/README.md

```diff
@@ -0,0 +1,44 @@
+# Creating "story" issues
+
+## Decision
+
+For larger features, for which it seems reasonable to create more than one issue, we will create overarching issues which 
+we'll name "story issues". Every story issues has several sub-issues a.k.a sub-tasks.
+
+## Rationale
+
+There is no clear limit or rule the specifies the size at which issues need to be broken into sub-issues, rather, 
+this should be done by intuition. Developers are expected to have an intrinsic understanding of the natural boundaries
+of features. In case of doubt it is better to create several small issues and list them in an overarching
+story issue. 
+
+Sub-issues may initially just be placeholders. That shows readers how the story is intended to be implemented and 
+which aspects have been taken into account, even if certain details might still be missing.
+
+Finally, having small, focused issues and PRs makes them more digestible, which in turn will benefit the review process.  
+
+## Approach
+
+Issues should be broken up if
+- there is more than one area of code involved (e.g. API, store,...)
+- work streams can be parallelized
+- there are orthogonal concerns
+
+The overarching story issues should contain a bulleted list with references to all sub-issues, for example:
+```markdown
+- [ ] #123
+- [ ] #456
+...
+```
+
+This will cause GitHub to automatically track the sub-issues in the overarching story issue, and it will display a progress
+indicator on overview pages. More information can be found in the [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists).
+
+For example, a story to add an API endpoint to the codebase could be split into the following sub-tasks:
```

Comment:
Maybe this is not the best example for an issue split, because I feel like an issue (and it's related PR) scope should be "a vertical slice" of the feature, so it should touch all the layers to have something functional after merging.
A better division could be: need to introduce API for a whole module like assets, it can be split in different issues like:
- add load asset endpoint
- add get all assets endpoint
- add get asset by id endpoint
- add delete asset by id endpoint
- etc...

This would guarantee orthogonality between issues and it will simplify PR reviews (no need to go through different PRs to review one of them)

Reply:
I don't disagree with you, but the point I wanted to make was less about _how to split_, than rather _to split at all_ and not to do everything in one single PR/Issue.

And to your point: if the requirement is to add _one single endpoint_ (as opposed to: several of them), what I proposed would be a way to go.
If the requirement is to add several endpoints, your proposal seems equally valid.

Reply:
OK, good point about the integration tests, i'll remove that. 
As to the other parts: separating them may make sense to create smaller, more focussed PRs, to satisfy time-constraints (e.g. API stub is needed first), to parallelize work, etc.

Reply:
Sure, but it will block the possibility to have an integration test until all the stories (or at least, the "code" ones) are implemented. Probably it's just about personal dev process, I am "one of those" that prefer to have a large scope test just before writing any production code

Reply:
ok, lets not get bogged down with this particular example, as it is not really relevant. I'll come up with another one.